specs/vep-1739-multiple-python-versions/README.md

```diff
@@ -145,9 +145,41 @@ This is not optimal, as the only way to change the vdk and base images is to cha
 As mentioned in the previous sections, we want to make vdk and job base images configurable per data job deployment, not Control Service one.
 To facilitate this, we need a configuration what python versions and respective vdk and job base images are supported by the Control Service.
 
-To configure these, a few options were explored:
+To configure these, a few options were explored. Below, we present the chosen solution, while the alternative options that were considered can be seen in the [Alternatives](#alternatives) section.
 
-#### Option 1: Introduce three environment variables, SUPPORTED_PYTHON_VERSIONS, JOB_BASE_IMAGE_TEMPLATE, and JOB_VDK_IMAGE_TEMPLATE.
+
+#### Use dictionaries to map supported python versions to corresponding base and vdk images.
+This option would utilize nested yaml dictionaries for configuration to map the supported version to a base and vdk images.
+
+In the helm chart values file, this configuration would be set like:
+```yaml
+supportedPythonVersions:
+   3.7:
+      baseImage: "registry.hub.docker.com/versatiledatakit/data-job-base-python-3.7-latest"
+      vdkImage: "registry.hub.docker.com/versatiledatakit/quickstart-vdk:release-3.7"
+   3.8:
+      baseImage: "registry.hub.docker.com/versatiledatakit/data-job-base-python-3.8-latest"
+      vdkImage: "registry.hub.docker.com/versatiledatakit/quickstart-vdk:release-3.8"
+   3.9:
+      baseImage: "registry.hub.docker.com/versatiledatakit/data-job-base-python-3.9-latest"
+      vdkImage: "registry.hub.docker.com/versatiledatakit/quickstart-vdk:release-3.9"
+```
+And in the application properties files, these could be set as a single property as:
+```shell
+datajobs.deployment.supportedPythonVersions={'3.7':{'baseImage': 'registry.hub.docker.com/versatiledatakit/data-job-base-python-3.7-latest', 'vdkImage': 'registry.hub.docker.com/versatiledatakit/quickstart-vdk:release-3.7'}, '3.8':{'baseImage': 'registry.hub.docker.com/versatiledatakit/data-job-base-python-3.8-latest', 'vdkImage': 'registry.hub.docker.com/versatiledatakit/quickstart-vdk:release-3.8'}, '3.9':{'baseImage': 'registry.hub.docker.com/versatiledatakit/data-job-base-python-3.9-latest', 'vdkImage': 'registry.hub.docker.com/versatiledatakit/quickstart-vdk:release-3.9'}}
+```
+**NOTE:** As the supported versions would be dynamically set, the application properties files would need to be manually updated accordingly.
+
+This option was chosen, as it would be relatively straightforward to implement in the helm chart of the Control Service, which would make it easily configurable by administrators.
```

Comment:
What are the configuration use cases or scenarios covered?
I would like to emphasise on API that is user-facing, and its usability.

Reply:
Will be added with a subsequent PR.